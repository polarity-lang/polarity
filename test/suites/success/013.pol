data Nat {
    Z,
    S(n: Nat),
}

let add(x y: Nat): Nat {
    x.match {
        Z => y,
        S(x') => ?,
    }
}
