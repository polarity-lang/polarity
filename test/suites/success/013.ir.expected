let add(x, y) {
    x.match {
        Z => y,
        S(x') => panic!("not yet implemented"),
    }
}
